### PR TITLE
feat: version update check

### PR DESCRIPTION
**Pull request by thisisnithin**
## Motivation and Context

This introduces a update check for the latest version of CLI on running a command. The cooldown is 24 hours. It can be disabled by setting env variable `DISABLE_UPDATE_CHECK=true`.

image

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: cli/src/utils.ts [hunk 1]
```diff
@@ -73,3 +76,48 @@ export function checkAPIKey() {
...
+    );
+
+    const newData = yaml.dump({
+      ...configFileData,
+      lastUpdateCheck: currentTime,
+    });
+    mkdirSync(configDir, { recursive: true });
+    writeFileSync(configFile, newData);
+  } catch {
+    console.log(pc.red('Failed to check for updates'));
```
Comment (StarpTech) [inline cli/src/utils.ts:121]: We should throw with an error so they can open a proper report on github.

File: cli/src/utils.ts [hunk 2]
```diff
@@ -73,3 +76,48 @@ export function checkAPIKey() {
...
+        borderColor: 'yellow',
+        borderStyle: 'round',
+      }),
+    );
+
+    const newData = yaml.dump({
+      ...configFileData,
+      lastUpdateCheck: currentTime,
+    });
+    mkdirSync(configDir, { recursive: true });
```
Comment (StarpTech) [inline cli/src/utils.ts:118]: Please put this into a "pre" config action.

File: cli/src/utils.ts [hunk 3]
```diff
@@ -73,3 +76,48 @@ export function checkAPIKey() {
...
+    writeFileSync(configFile, newData);
```
Comment (StarpTech) [inline cli/src/utils.ts:119]: Please also provide helper methods to write and read from the user config. That's a common use case.

File: cli/src/utils.ts [hunk 4]
```diff
@@ -73,3 +76,48 @@ export function checkAPIKey() {
...
+export const checkForUpdates = async () => {
+  try {
+    if (config.disableUpdateCheck === 'true') {
+      return;
+    }
+
+    const currentTime = Date.now();
+
+    const data = existsSync(configFile) ? yaml.load(readFileSync(configFile, 'utf8')) ?? {} : {};
+    const configFileData = JSON.parse(JSON.stringify(data));
```
Comment (StarpTech) [inline cli/src/utils.ts:89]: We should deal with the config in a type-safe way.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: cli/src/utils.ts (hunk 1 above)
Comment (thisisnithin) [reply cli/src/utils.ts]: Done

File: cli/src/utils.ts (hunk 2 above)
Comment (thisisnithin) [reply cli/src/utils.ts]: Done

File: cli/src/utils.ts (hunk 3 above)
Comment (thisisnithin) [reply cli/src/utils.ts]: Done

File: cli/src/utils.ts (hunk 4 above)
Comment (thisisnithin) [reply cli/src/utils.ts]: Done

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**